conans/model/profile.py

```diff
@@ -21,6 +22,8 @@ def loads(text):
 
         for setting in doc.settings.split("\n"):
             if setting:
+                if "=" not in setting:
+                    raise ConanException("Invalid setting line '%s'" % setting)
                 name, value = setting.split("=")
```

Comment:
You better strip both name and value, for people typing: "a = 3 " (both spaces around = and at the end)